test/Riok.Mapperly.Tests/Mapping/UserMethodTest.cs

```diff
@@ -78,6 +78,71 @@ public void WithExistingInstance()
             .HaveSingleMethodBody("target.StringValue = source.StringValue;");
     }
 
+    [Fact]
+    public void WithExistingInstanceNullable()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "partial void Map(A? source, B? target)",
+            "class A { public string StringValue { get; set; } }",
+            "class B { public string StringValue { get; set; } }");
+
+        TestHelper.GenerateMapper(source)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                    if (source == null || target == null)
+                        return;
+                    target.StringValue = source.StringValue;
+                """);
+    }
+
+    [Fact]
+    public Task WithExistingInstanceDisabledNullable()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "partial void Map(A source, B target)",
+            "class A { public string StringValue { get; set; } }",
+            "class B { public string StringValue { get; set; } }");
+
+        return TestHelper.VerifyGenerator(source, TestHelperOptions.DisabledNullable);
+    }
+
+    [Fact]
+    public void WithExistingCollectionInstance()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "partial void Map(IEnumerable<A> source, ICollection<B> target)",
+            "record A(string Value);",
+            "record B(string Value);");
+
+        TestHelper.GenerateMapper(source)
+            .Should()
+            .HaveMapMethodBody("""
+                    foreach (var item in source)
+                    {
+                        target.Add(MapToB(item));
+                    }
+                """);
+    }
+
+    [Fact]
+    public void WithExistingDictionaryInstance()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "partial void Map(IReadOnlyDictionary<string, A> source, IDictionary<string, B> target)",
+            "record A(string Value);",
+            "record B(string Value);");
+
+        TestHelper.GenerateMapper(source)
+            .Should()
+            .HaveMapMethodBody("""
+                    foreach (var item in source)
+                    {
+                        target.Add(item.Key, MapToB(item.Value));
```

Comment:
I think it would be safer to use the indexer instead of `.Add`, otherwise it could lead to collisions if the target already contains entries. This is an edge case IMO, could be cleaned up separately.

Reply:
Improved.